File: core/src/main/java/com/adobe/aem/guides/wknd/spa/angular/core/models/TransportListComponent.java
package com.adobe.aem.guides.wknd.spa.angular.core.models;

import com.adobe.cq.export.json.ComponentExporter;

import java.io.IOException;
import java.net.URISyntaxException;

public interface TransportListComponent extends ComponentExporter {
    public Transport[] getTransportList() throws IOException, URISyntaxException;
}

File: ui.frontend/src/app/components/transport-list/transport-list.component.ts
import { Component, Input, OnInit } from '@angular/core';
import {MapTo} from '@adobe/cq-angular-editable-components';

interface Transport {
    name: string;
    address: string;
    dept: string;
    type: string;
}

const CustomEditConfig = {
  emptyLabel: 'Transport List',
  isEmpty: cqModel =>
    !cqModel || !cqModel.transportList
};

@Component({
  selector: 'app-custom',
  templateUrl: './transport-list.component.html',
  styleUrls: ['./transport-list.component.css']
})
export class TransportListComponent implements OnInit {

  @Input() transportList: Transport[];

  constructor() { }

  ngOnInit(): void {
  }

}

MapTo('wknd-spa-angular/components/transport-list')(TransportListComponent, CustomEditConfig);
